memcached.c

```diff
@@ -3221,7 +3221,7 @@ static void drive_machine(conn *c) {
             bool reject;
             if (settings.maxconns_fast) {
                 STATS_LOCK();
-                reject = stats_state.curr_conns + stats_state.reserved_fds >= settings.maxconns - 1;
+                reject = sfd >= settings.maxconns - 1;
```

Comment:
neat! good point. I'll have to give this some deeper thought/testing myself before we merge but on a first pass it seems solid.

since maxconns can't be changed at runtime, we can probably move the STATS_LOCK(); back around just the rejected_conns++ (inside the if). Then there's no lock on connect again.

Reply:
moved the lock inside the if block.